paasta_tools/vitesscluster_tools.py

```diff
@@ -1,27 +1,550 @@
+            if cluster not in mysql_port_mappings:
+                log.error(
+                    f"MySQL Cluster {cluster} not found in system paasta config mysql_port_mappings"
+                )
+            if tablet_type not in mysql_port_mappings[cluster]:
+                continue
+            port = mysql_port_mappings[cluster][tablet_type]
+
+            # We use migration_replication delay for migration tablets and read_replication_delay for everything else
+            # Also throttling threshold for migration tablets is 2 hours, refresh and sanitized primaries at 30 seconds and everything else at 3 seconds
+            if tablet_type == "migration":
+                throttle_query_table = "migration_replication_delay"
+                throttle_metrics_threshold = "7200"
+            else:
+                throttle_query_table = "read_replication_delay"
+                if cluster.startswith("refresh") or cluster.startswith("sanitized"):
+                    throttle_metrics_threshold = "30"
+                else:
+                    throttle_metrics_threshold = "3"
+
+            tablet_pools.extend(
+                [
+                    get_tablet_pool_config(
+                        cell,
+                        db_name,
+                        keyspace,
+                        port,
+                        paasta_pool,
+                        paasta_cluster,
+                        zk_address,
+                        throttle_query_table,
+                        throttle_metrics_threshold,
+                        tablet_type,
+                        region,
+                        vttablet_resources,
+                    )
+                    for cell in cells
+                ]
+            )
+
+        config.append(
+            {
+                "name": keyspace,
+                "durabilityPolicy": "none",
+                "turndownPolicy": "Immediate",
+                "partitionings": [
+                    {
+                        "equal": {
+                            "parts": 1,
+                            "shardTemplate": {
+                                "databaseInitScriptSecret": {
+                                    "volumeName": "keyspace-fake-init-script",
+                                    "key": "/etc/init_db.sql",
+                                },
+                                "tabletPools": tablet_pools,
+                            },
+                        }
+                    }
+                ],
+            }
+        )
+
+    return config
+
+
 class VitessDeploymentConfigDict(LongRunningServiceConfigDict, total=False):
-    replicas: int
+    images: Dict[str, str]
+    cells: List[str]
+    region: str
+    zk_address: str
+    vtgate_resources: Dict[str, str]
+    vtadmin_resources: Dict[str, str]
+    vtctld_resources: Dict[str, str]
+    keyspaces: List[Dict[str, Any]]
+    updateStrategy: Dict[str, str]
+    paasta_cluster: str
```

Comment:
hmm, i just saw in the example config that we're passing in a paasta_cluster key: we shouldn't need to allow this to be user-specified (iirc, we'll already have the right value for that in `self.cluster`)

Reply:
(and we can probably generate the correct region in code rather than have that be user-specified - that will stop folks from making mistakes :))

Reply:
https://github.com/Yelp/paasta/pull/3889/commits/48ebe609e02377336248dcbb9bfed29a0d6213ae